crates/client/rpc-core/src/utils.rs

```diff
@@ -229,51 +229,52 @@ pub fn to_declare_transaction(
-            })
+                sender_address: declare_tx_v1.sender_address.into(),
+            });
+            Ok((tx, contract_class))
         }
         BroadcastedDeclareTransaction::V2(declare_tx_v2) => {
-            let signature = declare_tx_v2
-                .signature
-                .iter()
-                .map(|f| (*f).into())
-                .collect::<Vec<Felt252Wrapper>>()
-                .try_into()
-                .map_err(|_| BroadcastedTransactionConversionErrorWrapper::SignatureBoundError)?;
+            let signature = declare_tx_v2.signature.iter().map(|f| (*f).into()).collect::<Vec<Felt252Wrapper>>();
 
             let casm_constract_class = flattened_sierra_to_casm_contract_class(declare_tx_v2.contract_class.clone())
-                .map_err(|_| BroadcastedTransactionConversionErrorWrapper::SierraCompilationError)?;
-            let contract_class = ContractClassV1::try_from(casm_constract_class.clone())
-                .map_err(|_| BroadcastedTransactionConversionErrorWrapper::CasmContractClassConversionError)?;
+                .map_err(|_| BroadcastedTransactionConversionError::SierraCompilationError)?;
 
             // ensuring that the user has signed the correct class hash
             if get_casm_cotract_class_hash(&casm_constract_class) != declare_tx_v2.compiled_class_hash {
-                return Err(BroadcastedTransactionConversionErrorWrapper::CompiledClassHashError);
+                return Err(BroadcastedTransactionConversionError::CompiledClassHashError);
             }
 
-            Ok(DeclareTransaction {
-                version: 2_u8,
-                sender_address: declare_tx_v2.sender_address.into(),
-                nonce: Felt252Wrapper::from(declare_tx_v2.nonce),
-                max_fee: Felt252Wrapper::from(declare_tx_v2.max_fee),
+            let contract_class = BlockifierContractClass::V1(
+                ContractClassV1::try_from(casm_constract_class)
+                    .map_err(|_| BroadcastedTransactionConversionError::CasmContractClassConversionError)?,
+            );
+
+            let tx = DeclareTransaction::V2(DeclareTransactionV2 {
+                max_fee: declare_tx_v2
+                    .max_fee
+                    .try_into()
+                    .map_err(|_| BroadcastedTransactionConversionError::InvalidMaxFee)?,
                 signature,
-                contract_class: BlockifierContractClass::V1(contract_class),
-                compiled_class_hash: Some(Felt252Wrapper::from(declare_tx_v2.compiled_class_hash)),
+                nonce: declare_tx_v2.nonce.into(),
                 class_hash: declare_tx_v2.contract_class.class_hash().into(),
-                is_query: declare_tx_v2.is_query,
-            })
+                sender_address: declare_tx_v2.sender_address.into(),
+                compiled_class_hash: declare_tx_v2.compiled_class_hash.into(),
+            });
+            Ok((tx, contract_class))
```

Comment:
Actually this code should be removed. This logic lives in `crates/primitives/starknet/src/transaction/from_broadcasted_transactions.rs` now. As a `TryInto` impl

Reply:
And you are right. We should use the `is_query` to augment the nonce during the TryInto

Reply:
It is still possible to do fee estimate.
Because the runtime knows it is in the `fee_estimate` context, it will call the `tx.execute` methode with `is_query` set at `true` and the `get_account_transaction_context` impl will make sure to offset the version number